src/backend/web/validation.js

```diff
@@ -73,6 +73,13 @@ const queryValidationRules = [
     .isIn(['post', 'author'])
     .withMessage('invalid filter value')
     .bail(),
+
+  query('perPage', 'per_page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+  query('page', 'page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
```

Comment:
Also, you need to add more tests for the validation on these routes to make sure that good values for `page` and `per_page` give 200s, and values outside the valid range return 400s.

Reply:
They should give the right response codes now according to the values